### PR TITLE
[MM-13147] Fix for file_upload crash when mimeTypesSuffix is not present

### PR DESCRIPTION
**Pull request by sudheerDev**
#### Summary
Fix for file_upload crash for certain mimeTypes

#### Ticket Link
[MM-13147](https://mattermost.atlassian.net/browse/MM-13147)

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Added or updated unit tests (required for all new features)

### CODE REVIEW

File: utils/file_utils.test.jsx [hunk 1]
```diff
@@ -104,6 +104,10 @@ describe('FileUtils.canUploadFiles', () => {
         it('mime type for unknown file format', () => {
             assert.equal(getFileTypeFromMime('application/unknownFormat'), 'other');
         });
+
+        it('mime type for no suffix', () => {
+            assert.equal(getFileTypeFromMime('asdasd'), 'other');
+        });
```
Comment (saturninoabril) [inline utils/file_utils.test.jsx]: 💯

**Comment by enahum** [comment]
@sudheerDev build failed

File: utils/file_utils.jsx [hunk 2]
```diff
@@ -64,20 +64,25 @@ export function getFileTypeFromMime(mimetype) {
...
-        return 'pdf';
-    } else if (mimeTypeSuffix.includes('vnd.ms-excel') || mimeTypeSuffix.includes('spreadsheetml') || mimeTypeSuffix.includes('vnd.sun.xml.calc') || mimeTypeSuffix.includes('opendocument.spreadsheet')) {
-        return 'spreadsheet';
-    } else if (mimeTypeSuffix.includes('vnd.ms-powerpoint') || mimeTypeSuffix.includes('presentationml') || mimeTypeSuffix.includes('vnd.sun.xml.impress') || mimeTypeSuffix.includes('opendocument.presentation')) {
-        return 'presentation';
-    } else if ((mimeTypeSuffix === 'msword') || mimeTypeSuffix.includes('vnd.ms-word') || mimeTypeSuffix.includes('officedocument.wordprocessingml') || mimeTypeSuffix.includes('application/x-mswrite')) {
-        return 'word';
+    }
+
+    if(mimeTypeSuffix) {
```
Comment (enahum) [inline utils/file_utils.jsx]:
```suggestion
    if (mimeTypeSuffix) {
```

File: utils/file_utils.jsx [hunk 3]
```diff
@@ -64,20 +64,25 @@ export function getFileTypeFromMime(mimetype) {
...
+        if (mimeTypeSuffix === 'pdf') {
+             return 'pdf';
```
Comment (enahum) [inline utils/file_utils.jsx]:
```suggestion
            return 'pdf';
```

File: utils/file_utils.jsx [hunk 4]
```diff
@@ -64,20 +64,25 @@ export function getFileTypeFromMime(mimetype) {
...
+        } else if (mimeTypeSuffix.includes('vnd.ms-excel') || mimeTypeSuffix.includes('spreadsheetml') || mimeTypeSuffix.includes('vnd.sun.xml.calc') || mimeTypeSuffix.includes('opendocument.spreadsheet')) {
+             return 'spreadsheet';
```
Comment (enahum) [inline utils/file_utils.jsx]:
```suggestion
            return 'spreadsheet';
```

File: utils/file_utils.jsx [hunk 5]
```diff
@@ -64,20 +64,25 @@ export function getFileTypeFromMime(mimetype) {
...
+        } else if (mimeTypeSuffix.includes('vnd.ms-powerpoint') || mimeTypeSuffix.includes('presentationml') || mimeTypeSuffix.includes('vnd.sun.xml.impress') || mimeTypeSuffix.includes('opendocument.presentation')) {
+             return 'presentation';
```
Comment (enahum) [inline utils/file_utils.jsx]:
```suggestion
            return 'presentation';
```

File: utils/file_utils.jsx [hunk 6]
```diff
@@ -64,20 +64,25 @@ export function getFileTypeFromMime(mimetype) {
...
+        } else if ((mimeTypeSuffix === 'msword') || mimeTypeSuffix.includes('vnd.ms-word') || mimeTypeSuffix.includes('officedocument.wordprocessingml') || mimeTypeSuffix.includes('application/x-mswrite')) {
+             return 'word';
```
Comment (enahum) [inline utils/file_utils.jsx]:
```suggestion
            return 'word';
```

**Pull request merged**